Source/Library/GameBoardWindow/TokenTile.cpp

```diff
@@ -0,0 +1,72 @@
+#include <QPushButton>
+#include "TicTacToe/Library/GameBoardWindow/TokenTile.hpp"
+
+using namespace boost;
+using namespace boost::signals2;
+using namespace TicTacToe;
+
+TokenTile::TokenTile(QWidget* parent)
+  : TokenTile(Board::Token::NONE, parent) {
+}
+
+TokenTile::TokenTile(Board::Token token, QWidget* parent)
+  : QWidget(parent)
+  , m_token(token) {
+  setFixedSize(100, 100);
+  auto button = new QPushButton(this);
+  button->setGeometry(0, 0, width(), height());
+  set_token(token);
+  QObject::connect(button, &QPushButton::clicked,
```

Comment:
You should be able to forward the signal exactly:

    QObject::connect(button, &QPushButton::clicked, this, m_clickedSignal);

Reply:
It doesn't work. The compiler complains "C:\Users\Shaobin\source\repos\tic_tac_toe\Source\Library\GameBoardWindow\TokenTile.cpp(18,1): error C2280: 'boost::signals2::signal<void (void),boost::signals2::optional_last_value<void>,int,std::less<int>,boost::function<void (void)>,boost::function<R (const boost::signals2::connection &)>,boost::signals2::dummy_mutex>::signal(const boost::signals2::signal<void (void),boost::signals2::optional_last_value<void>,int,std::less<int>,boost::function<void (void)>,boost::function<R (const boost::signals2::connection &)>,boost::signals2::dummy_mutex> &)': attempting to reference a deleted function"

I can avoid this error by using the code below
QObject::connect(button, SIGNAL(QPushButton::clicked), this, SIGNAL(m_clickedSignal));
or 
my original code with lambda expression.

Reply:
Okay keep it the way you had it originally.

Reply:
So which way do you prefer in our project? Using a lambda expression or a SIGNAL macro?